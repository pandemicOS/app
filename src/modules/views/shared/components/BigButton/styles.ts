import { StyleSheet } from 'react-native';

import { STYLE } from '../../../../../constants';

export default StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'center',
    backgroundColor: STYLE.COLOR.WHITE,
    borderRadius: 4,
    borderColor: STYLE.COLOR.PRIMARY,
    borderWidth: 2,
    height: 120,
    shadowColor: STYLE.COLOR.BLACK,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 2,
    elevation: 2
  },
  wrapper: {
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center'
  },
  title: {
    fontSize: 21,
    color: STYLE.COLOR.BLACK,
    alignSelf: 'center',
    fontWeight: STYLE.FONT.WEIGHTS.MEDIUM,
  },
  subtitle: {
    marginTop: '2%',
    marginHorizontal: 10,
    fontSize: 14,
    color: STYLE.COLOR.BLACK,
    alignSelf: 'center',
    fontWeight: STYLE.FONT.WEIGHTS.REGULAR,
    textAlign: 'center',
  }
});
