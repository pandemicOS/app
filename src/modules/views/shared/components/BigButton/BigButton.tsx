import React from 'react';
import { View, Text, TouchableOpacity, StyleProp, ViewStyle } from 'react-native';

import styles from './styles';

export interface IBigButtonProps {
  title: string;
  subtitle?: string;
  disabled?: boolean;
  onPress?: () => void;
  style?: StyleProp<ViewStyle>;
}

export default class BigButton extends React.PureComponent<IBigButtonProps> {
  public render() {
    const { title, disabled, subtitle, onPress, style } = this.props;
    return (
      <TouchableOpacity
        style={[styles.container, style, { opacity: disabled ? 0.5 : 1 }]}
        onPress={onPress}
        disabled={disabled}
      >
        <View style={styles.wrapper}>
          <Text style={styles.title}>{title}</Text>
          {subtitle ? <Text style={styles.subtitle}>{subtitle}</Text> : null}
        </View>
      </TouchableOpacity>
    );
  }
}
