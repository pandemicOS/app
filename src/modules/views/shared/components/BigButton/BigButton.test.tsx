import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import BigButton, { IBigButtonProps } from './BigButton';

describe('BigButton', () => {
  let props: IBigButtonProps;
  beforeEach(() => {
    props = {
      title: '',
      subtitle: 'subtitle',
      onPress: jest.fn(),
      disabled: false,
    };
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(<BigButton {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders disabled', () => {
    props.disabled = true;
    const rendered = renderer.create(<BigButton {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders without subtitle', () => {
    props.subtitle = null;
    const rendered = renderer.create(<BigButton {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should handle onPress', async () => {
    const wrapper = shallow(<BigButton {...props} />);
    await (wrapper.find('TouchableOpacity').first().prop('onPress') as any)();
    expect(props.onPress).toBeCalled();
  });

  it('should render without onPress', async () => {
    props.onPress = undefined;
    const rendered = renderer.create(<BigButton {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
