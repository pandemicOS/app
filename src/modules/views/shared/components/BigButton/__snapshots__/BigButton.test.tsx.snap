// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigButton renders disabled 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 4,
      "elevation": 2,
      "height": "37%",
      "justifyContent": "center",
      "opacity": 0.5,
      "shadowColor": "#1E272E",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#1E272E",
          "fontSize": 21,
          "fontWeight": "600",
        }
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#1E272E",
          "fontSize": 13,
          "fontWeight": "300",
          "marginTop": "2%",
        }
      }
    >
      subtitle
    </Text>
  </View>
</View>
`;

exports[`BigButton renders without crashing 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 4,
      "elevation": 2,
      "height": "37%",
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#1E272E",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#1E272E",
          "fontSize": 21,
          "fontWeight": "600",
        }
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#1E272E",
          "fontSize": 13,
          "fontWeight": "300",
          "marginTop": "2%",
        }
      }
    >
      subtitle
    </Text>
  </View>
</View>
`;

exports[`BigButton renders without subtitle 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 4,
      "elevation": 2,
      "height": "37%",
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#1E272E",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#1E272E",
          "fontSize": 21,
          "fontWeight": "600",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`BigButton should render without onPress 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 4,
      "elevation": 2,
      "height": "37%",
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#1E272E",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#1E272E",
          "fontSize": 21,
          "fontWeight": "600",
        }
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#1E272E",
          "fontSize": 13,
          "fontWeight": "300",
          "marginTop": "2%",
        }
      }
    >
      subtitle
    </Text>
  </View>
</View>
`;
